eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/io/AccessionedVariantMongoReader.java

```diff
@@ -169,25 +171,31 @@ public void open(ExecutionContext executionContext) throws ItemStreamException {
                                                                      submittedVariantValidated, allelesMatch,
                                                                      assemblyMatch, evidence);
 
-            String variantId = (contig + "_" + start + "_" + reference + "_" + alternate).toUpperCase();
-            if (variants.containsKey(variantId)) {
-                variants.get(variantId).addSourceEntry(sourceEntry);
-            } else {
-                Variant variant = new Variant(contig, start, end, reference, alternate);
-                variant.setMainId(buildId(rs));
-                variant.addSourceEntry(sourceEntry);
-                variants.put(variantId, variant);
-            }
+            addToVariants(variants, contig, start, rs, reference, alternate, sourceEntry);
         }
         return new ArrayList<>(variants.values());
     }
 
-    private long calculateEnd(String reference, String alternate, long start) {
+    protected void addToVariants(Map<String, Variant> variants, String contig, long start, long rs, String reference,
+                               String alternate, VariantSourceEntry sourceEntry) {
+        String variantId = (contig + "_" + start + "_" + reference + "_" + alternate).toUpperCase();
```

Comment:
This is a one-liner, but extracting it to a separate method would allow to simplify the `addToVariants` interface and make it more obvious, as in:

```java
protected void addToVariants(Map<String, Variant> variants, String variantId, VariantSourceEntry sourceEntry)
```

Reply:
no, the `start` and `ref` and `alt` are used in the rest of the addToVariants method (the `else` part)

Reply:
unless you want to split the id by underscore to retrieve those fields againg, which I think is a bad idea